bricks/stm32/common.ld

```diff
@@ -70,7 +70,24 @@ SECTIONS
         _ebss = .;         /* define a global symbol at bss end; used by startup code */
     } >RAM
 
-    _heap_start = _ebss;   /* heap starts just after statically allocated memory */
+    /* The user_ram section between ebss and sstack is used for two purposes:
+       - The system uses the first portion to store user program data.
+       - The system can then run an application, which has read-only access to
+         the user data. The remaining space is used by the application as heap.
+    */
+    _pbsys_program_load_user_ram_start = _ebss;
+    .user_ram :
+    {
+        . = ALIGN(4);
+        /**
+         * The full storage may be loaded to RAM on boot, so we need to have
+         * at least as much RAM available. We also want to ensure there is
+         * at least 2K of additional RAM for the application heap.
+         */
+        . = . + _pbdrv_block_device_storage_size + 2048;
```

Comment:
I considered it more like an emergency minimum so that the out of memory exception can be raised without crashing, just like we have that 1K of spare stack before overflow.

But since the available RAM is much larger on all hubs than we'll allow for storage space, there is always going to be a substantially larger amount of heap available.

So if we add another per-hub configuration value, it wouldn't be an independent variable.

Reply:
We could set it to 4K though as a better minimum.

For perspective, there is about 280K available on Prime Hub and storage is set to 256K, so there's at least 34K heap.

Reply:
Set to 4K.